pkgs/tools/misc/hdf4/default.nix

```diff
@@ -31,11 +31,12 @@ stdenv.mkDerivation rec {
     "-DHDF4_BUILD_WITH_INSTALL_NAME=OFF"
     "-DHDF4_ENABLE_JPEG_LIB_SUPPORT=ON"
     "-DHDF4_ENABLE_NETCDF=OFF"
-    "-DHDF4_ENABLE_SZIP_ENCODING=ON"
-    "-DHDF4_ENABLE_SZIP_SUPPORT=ON"
     "-DHDF4_ENABLE_Z_LIB_SUPPORT=ON"
     "-DHDF4_BUILD_FORTRAN=OFF"
     "-DJPEG_DIR=${libjpeg}"
+  ] ++ stdenv.lib.optionals (szip != null) [
+    "-DHDF4_ENABLE_SZIP_ENCODING=ON"
```

Comment:
I looked at `szip` homepage and the prohibition for use of `szip` (without a special license) for commercial use involving encoding is presented as a _licensing_ restriction there. We generally believe project homepages when they claim a licensing restriction exists.

Also, the license itself says it is revocable. And it contains «Do not modify or use for other purposes». In NixPkgs, such a license is considered `unfree`.

So, I do not see what is the problem with the current setting: by default, a sensible free build is performed, and _if_ the user overrides by passing actual `szip` instead of null, `szip` is used.